x-pack/test/api_integration/apis/endpoint/whitelist.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect/expect.js';
+import { Client } from 'elasticsearch';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function({ getService }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertest');
+  const es: Client = getService('legacyEs');
+
+  const nextPrevPrefixDateRange = "date_range=(from:'2018-01-10T00:00:00.000Z',to:now)";
+  const nextPrevPrefixSort = 'sort=@timestamp';
+  const nextPrevPrefixOrder = 'order=desc';
+  const nextPrevPrefixPageSize = 'page_size=10';
+  const nextPrevPrefix = `${nextPrevPrefixDateRange}&${nextPrevPrefixSort}&${nextPrevPrefixOrder}&${nextPrevPrefixPageSize}`;
+
+  const sizeOfFixture: number = 1;
+
+  describe('test whitelist api', () => {
+    describe('Tests for whitelist API', () => {
+      beforeEach(() => esArchiver.load('endpoint/whitelist/api_feature'));
+      afterEach(() => esArchiver.unload('endpoint/whitelist/api_feature'));
+
+      it('should return whitelist rules when the GET method is called', async () => {
+        const resp = await supertest.get('/api/endpoint/whitelist').set('kbn-xsrf', 'xxx');
+
+        expect(resp.statusCode).to.equal(200);
+        const responseBody = JSON.parse(resp.text);
+        expect(responseBody.length).to.equal(sizeOfFixture);
+      });
+
+      it('should insert a whilelist rule into elasticsearch properly', async () => {
+        await supertest
+          .post('/api/endpoint/whitelist')
+          .set('kbn-xsrf', 'xxx')
+          .send({ alert_id: '123', file_path: 'you havent seen yet' })
+          .then(function(res) {
+            expect(res.statusCode).to.equal(200);
+            const responseBody = JSON.parse(res.text);
+            expect(responseBody.length).to.equal(1);
+          })
+          .then(async function(resp) {
+            await supertest
+              .get('/api/endpoint/whitelist')
+              .set('kbn-xsrf', 'xxx')
+              .then(function(getResp) {
+                const getRespBody = JSON.parse(getResp.text);
+                expect(getRespBody.length).to.equal(sizeOfFixture + 1);
+              });
+          });
+      });
+
+      it('should insert multiple whilelist rules into elasticsearch from a single request properly', async () => {
```

Comment:
this is different usage I think, than we have in the old Endgame SMP where one whitelist entry could / would contain multiple keys in the same filter if they were all submitted in the same time in the UI or API.  is this an intentional change to keep the keys simpler or for any other reason?

Reply:
I'm actually not quite sure. I was under the impression it should create multiple rules. @ferullo is this the case?

Reply:
Yeah, each whitelist entry should be a different rule in the rule list. If we get a copy of the whitelist artifact this would product for multiple rules I can confirm it looks right.

Reply:
I sent us on a wild goose chase on this one, I validated in old SMP and its creating 2 rule entries in the whitelist, sorry.  Not sure where / what I was thinking of, except maybe advanced whitelisting which we aren't talking about.

Reply:
Fantastic. thanks for checking on this; I wasn't sure either.